test/osb_and_plugin_test/plugin_test/platform_termination_plugin_test.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/gofrs/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"net/http"
+)
+
+var _ = Describe("Service Manager Platform Termination Plugin Tests", func() {
+	var (
+		ctx          *common.TestContext
+		brokerServer *common.BrokerServer
+		osbURL       string
+
+		brokerID  string
+		serviceID string
+		planID    string
+	)
+
+	JustBeforeEach(func() {
+		username, password := test.RegisterBrokerPlatformCredentials(ctx.SMWithBasic, brokerID)
+		ctx.SMWithBasic.SetBasicCredentials(ctx, username, password)
+	})
+
+	AfterEach(func() {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+	})
+
+	Describe("Platform termination OSB plugin", func() {
+
+		BeforeEach(func() {
+			ctx = common.NewTestContextBuilderWithSecurity().Build()
+			UUID, err := uuid.NewV4()
+			Expect(err).ToNot(HaveOccurred())
+			planID = UUID.String()
+			plan1 := common.GenerateTestPlanWithID(planID)
+			UUID, err = uuid.NewV4()
+			Expect(err).ToNot(HaveOccurred())
+			serviceID = UUID.String()
+			service1 := common.GenerateTestServiceWithPlansWithID(serviceID, plan1)
+			catalog := common.NewEmptySBCatalog()
+			catalog.AddService(service1)
+
+			brokerID, _, brokerServer = ctx.RegisterBrokerWithCatalog(catalog).GetBrokerAsParams()
+			brokerServer.ShouldRecordRequests(true)
+			common.CreateVisibilitiesForAllBrokerPlans(ctx.SMWithOAuth, brokerID)
+			osbURL = "/v1/osb/" + brokerID
+
+			ctx.SMWithBasic.PUT(osbURL+"/v2/service_instances/12345").
+				WithHeader("Content-Type", "application/json").
+				WithJSON(object{"service_id": serviceID, "plan_id": planID}).
+				Expect().Status(http.StatusCreated)
+
+			testPlatformID := ctx.TestPlatform.GetID()
+
+			err = ctx.SMRepository.InTransaction(context.TODO(), func(ctx context.Context, storage storage.Repository) error {
+				var updatedPlatform types.Object
+				byID := query.ByField(query.EqualsOperator, "id", testPlatformID)
+				platformFromStorage, err := storage.Get(ctx, types.PlatformType, byID)
+				Expect(err).ToNot(HaveOccurred())
+
+				platformFromStorage.(*types.Platform).Active = false
+				if updatedPlatform, err = storage.Update(ctx, platformFromStorage, types.LabelChanges{}); err != nil {
+					return err
+				}
+				Expect(updatedPlatform.(*types.Platform).Active).To(Equal(false))
+				return nil
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			ctx.SMWithOAuth.DELETE(web.PlatformsURL+"/"+testPlatformID).
+				WithQuery("cascade", "true").
+				Expect().
+				Status(http.StatusAccepted)
+		})
+
+		osbOperations := []struct {
+			name           string
+			method         string
+			path           string
+			queries        []string
+			expectedStatus int
+		}{
+			{"fetchCatalog", "GET", "/v2/catalog", []string{""}, http.StatusOK},
+			{"provision", "PUT", "/v2/service_instances/1234", []string{""}, http.StatusUnprocessableEntity},
+			{"provision", "PUT", "/v2/service_instances/1234", []string{"accepts_incomplete=true"}, http.StatusUnprocessableEntity},
```

Comment:
duplicate name - maybe name this one "provisionAsync" so we can tell the difference in the test output?

Reply:
doe